HW2/P2/parallel_vector.pyx

```diff
@@ -81,10 +81,20 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+
+           for idx in prange(src.shape[0], num_threads=4, schedule=dynamic):
+
                if counts[src[idx]] > 0:
-                   counts[dest[idx]] += 1
-                   counts[src[idx]] -= 1
+                   if src[idx] == dest[idx]:
+                       continue
+                   else:
```

Comment:
I think that if src[idx] == dest[idx] you still need to lock that position. What if an other thread has the same src[idx] (dest[idx])?